macrosynergy/pnl/historic_portfolio_volatility.py

```diff
@@ -60,7 +60,7 @@ def _weighted_covariance(
 
     xnans, ynans = np.isnan(x), np.isnan(y)
     wmask = xnans | ynans
-    weightslen = min(sum(~wmask), lback_periods)
+    weightslen = min(sum(~wmask), lback_periods if lback_periods > 0 else len(x))
```

Comment:
Do we need to be explicit about only -1 is full look back? So basically:

```python
assert lback_periods == -1 or lback_periods > 0
```

Just to insure the user don't accidentally put in -21 days or similar?

Reply:
This is asserted earlier at https://github.com/macrosynergy/macrosynergy/blob/d604e10c97a61d33744e86ba6611ef8e783a0482/macrosynergy/pnl/historic_portfolio_volatility.py#L205